### PR TITLE
Add CRTB Downstream Handler

### PR DESCRIPTION
**Pull request by JonCrowther**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #47660
* #47572
* __->__ #47456
* #47455

### CODE REVIEW

File: pkg/controllers/managementuser/rbac/roletemplates/crtb_handler.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			return nil, err
+		}
+	}
+
+	// If we didn't find an existing CRB, create it
+	if matchingCRB == nil {
+		if _, err := c.crbClient.Create(crb); err != nil {
+			return nil, err
+		}
+	}
```
Comment (aruiz14) [inline pkg/controllers/managementuser/rbac/roletemplates/crtb_handler.go:73]: Could this cause a race condition while access is temporarily lost due to removing CRB but the new ones were not created yet? Or what if the Creation fails for some reason?
For both reasons, I would move the cleanup to the end of the function (also, under which circumstances there could there be multiple CRBs that need to be removed?

File: pkg/controllers/managementuser/rbac/roletemplates/crtb_handler.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "crb-",
+			Labels:       map[string]string{ownerLabel: "true"},
+		},
+		RoleRef:  roleRef,
```
Comment (aruiz14) [inline pkg/controllers/managementuser/rbac/roletemplates/crtb_handler.go:130]: Nit-pick: I believe that to in-line these variables would make this more clear, with the exception of those fields that could find an error during their processing.

File: pkg/controllers/managementuser/rbac/roletemplates/register.go [hunk 3]
```diff
@@ -9,7 +9,9 @@ import (
 func Register(ctx context.Context, workload *config.UserContext) {
 	management := workload.Management.WithAgent("rbac-role-templates")
 
-	management.Management.ClusterRoleTemplateBindings("").AddLifecycle(ctx, "cluster-crtb-handler", newCRTBLifecycle(workload))
+	c := newCRTBHandler(workload)
+	management.Wrangler.Mgmt.ClusterRoleTemplateBinding().OnChange(ctx, "cluster-crtb-change-handler", c.OnChange)
+	management.Wrangler.Mgmt.ClusterRoleTemplateBinding().OnRemove(ctx, "cluster-crtb-remove-handler", c.OnRemove)
```
Comment (aruiz14) [inline pkg/controllers/managementuser/rbac/roletemplates/register.go:14]: Sorry we may have talked about this but my memories are vague: why should we move from Lifecycle to separate OnChange + `OnRemove`? Is there any difference in behavior?
Beware that existing objects may already have a finalizer for the previous `cluster-crtb-handler` Lifecycle handler, which would remain unhandled with this change. We need to either pro-actively remove it (in `OnChange`) or just make this use the older name.

**Comment by JonCrowther** [comment]
Closing in favour of https://github.com/rancher/rancher/pull/48114

**Pull request closed without merging**